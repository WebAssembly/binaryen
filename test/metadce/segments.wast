(module
 (import "env" "g1" (global $g1 i32))
 (import "env" "g2" (global $g2 i32))

 (table $tbl 1 funcref)
 (elem (offset (global.get $g1)) funcref (ref.func $f))

 (memory 3)
 (data (offset (global.get $g2)) "xxx")

 (export "f" (func $f))
 (func $f (param i32)
  (call_indirect (param i32) (local.get 0) (i32.load8_u (i32.const 0))))
)
