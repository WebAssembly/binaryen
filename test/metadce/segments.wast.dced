(module
 (type $0 (func (param i32)))
 (import "env" "g1" (global $g1 i32))
 (import "env" "g2" (global $g2 i32))
 (memory $0 3)
 (data $0 (global.get $g2) "xxx")
 (table $tbl 1 funcref)
 (elem $0 (global.get $g1) $f)
 (export "f" (func $f))
 (func $f (type $0) (param $0 i32)
  (call_indirect $tbl (type $0)
   (local.get $0)
   (i32.load8_u
    (i32.const 0)
   )
  )
 )
)
