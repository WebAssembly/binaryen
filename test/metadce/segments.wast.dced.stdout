unused: dseg$0$3
unused: eseg$0$4
